packages/client-sdk-nodejs/src/internal/vector-data-client.ts

```diff
@@ -0,0 +1,242 @@
+import {version} from '../../package.json';
+import {IVectorIndexDataClient} from '@gomomento/sdk-core/dist/src/internal/clients/vector/IVectorIndexDataClient';
+import {VectorIndexItem} from '@gomomento/sdk-core/dist/src/messages/vector-index';
+import {
+  CredentialProvider,
+  InvalidArgumentError,
+  MomentoLogger,
+  MomentoLoggerFactory,
+  SearchOptions,
+  VectorAddItemBatch,
+  VectorDeleteItemBatch,
+  VectorSearch,
+} from '@gomomento/sdk-core';
+import {VectorIndexClientProps} from '../vector-index-client-props';
+import {VectorIndexConfiguration} from '../config/vector-index-configuration';
+import {ChannelCredentials, Interceptor} from '@grpc/grpc-js';
+import {vectorindex} from '@gomomento/generated-types/dist/vectorindex';
+import {Header, HeaderInterceptorProvider} from './grpc/headers-interceptor';
+import {ClientTimeoutInterceptor} from './grpc/client-timeout-interceptor';
+import {cacheServiceErrorMapper} from '../errors/cache-service-error-mapper';
+import {
+  validateIndexName,
+  validateTopK,
+} from '@gomomento/sdk-core/dist/src/internal/utils';
+import {normalizeSdkError} from '@gomomento/sdk-core/dist/src/errors';
+
+export class VectorDataClient implements IVectorIndexDataClient {
```

Comment:
nit: should we call this`VectorIndexDataClient` as seems like we have added the `index` term everywhere else

Reply:
fair. will rename in a subsequent PR when I break up the control client.